Mathport/Syntax/Transform/Basic.lean

```diff
@@ -43,7 +43,7 @@ scoped elab "mathportTransformerList%" : term => do
 partial def applyTransformers (transformers : Array Transformer) (stx : Syntax) : M Syntax :=
   stx.rewriteBottomUpM fun stx => do
     for tr in transformers do
-      if let some stx' ← catchUnsupportedSyntax do tr stx then
+      if let some stx' ← catchUnsupportedSyntax do withRef stx do tr stx then
```

Comment:
is this `do` a weird way to spell `<|`?

Reply:
*Both* `do`s could be replaced by `<|` here:
```lean
if let some stx' ← catchUnsupportedSyntax <| withRef stx <| tr stx then
```

Not sure if that's an improvement though.

Reply:
nested `do` blocks make me nervous since they have complicated effects on control flow, so I would prefer to avoid them unless they actually do something. Also it is a lot easier to miss `do` than `<|` / `$` if you don't have syntax highlighting (like the second `do` in the presentation here on github).

Reply:
FWIW, the stdlib prefers `do` over `<|`, though I don't think I've seen it without a linebreak yet. There is a related TODO we never came back to about actually supporting `do` blocks nested in calls in a way that does not terminate the "`do effects", as long as the called function is sufficiently generic over the used monad. We couldn't even agree on a preferred syntax; simply doing it when the last argument is an unparenthesized `do` block might be too magical (or maybe it would be about the same magic level as other parts of Lean's syntax).